.github/test_on_push.yml

```diff
@@ -0,0 +1,55 @@
+name: pybamm_cookiecutter
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+    - main
+jobs:
+  style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - name: Check style
+        run: |
+          python -m pip install pre-commit
+          pre-commit run -a
+
+  template_test:
+    needs: style
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-12, macos-14, windows-latest]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+    name:
+      Template Tests (${{ matrix.os }} / Python ${{ matrix.python-version }})
+    steps:
+      - name: Checkout pybamm_cookiecutter
+        uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+
+      - name: Set up uv
+        uses: yezz123/setup-uv@v4
+        with:
+          uv-venv: ".venv"
```

Comment:
Question: is it possible to cache `uv` through this action? I haven't explored it. We should cache either `pip` or `uv`, since both of them store cached wheels in difference places, IIRC.

Reply:
Yes, makes sense – let's look at `uv`'s caching later during the project. As an aside, I did a quick run for `uv` in my PyBaMM repository locally and it works quite well now, saving up a minute of time and more.